test/library/packages/LinearAlgebra/correctness/testSparseTranspose.chpl

```diff
@@ -0,0 +1,81 @@
+use LinearAlgebra;
+use LinearAlgebra.Sparse;
+use BlockDist;
+use List;
+
+config const n = 16;
+config const stride = 7;
+
+// TODO: As of now distributed sparse is not supported, though it is
+//       likely to run fine with the current implementation of transpose.
+//       The `isSparseArr()` in the `where` clause prevents `transpose`
+//       being called on a distributed sparse domain/array.
+
+// Domains
+
+var parentDom2D = {1..n, 1..n};
+// var distParentDom2D = {1..n, 1..n} dmapped Block({1..n, 1..n});
+
+proc getSparse(d, stride) {
+  var sdom: sparse subdomain(d);
+  var indices: [1..0] (if d.rank==1 then d.idxType else d.rank*d.idxType);
+  var count = 0;
+  for ind in d {
+    if (count % stride == 0) then indices.push_back(ind);
```

Comment:
you can use the same strategy (i.e. `makeIndexBuffer`) to create the sparse domain here in the test.